tests/formatter_test.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+#
+# This software may be modified and distributed under the terms
+# of the MIT license.  See the LICENSE file for details.
+
+from logging import FileHandler, makeLogRecord
+import os
+import sys
+import unittest
+
+from logstash_async.formatter import LogstashFormatter
+
+
+class ExceptionCatchingFileHandler(FileHandler):
+    def __init__(self, *args, **kwargs):
+        FileHandler.__init__(self, *args, **kwargs)
+        self.exception = None
+
+    def handleError(self, record):
+        self.exception = sys.exc_info()
+
+
+class LogstashFormatterTest(unittest.TestCase):
+    def test_format(self):
+        file_handler = ExceptionCatchingFileHandler(os.devnull)
+        file_handler.setFormatter(LogstashFormatter(ensure_ascii=False))
+        file_handler.emit(makeLogRecord({"msg": "тест"}))
```

Comment:
The non-ASCII characters break on Python2.
Making the string a unicode string on Python2 and leave as is on Python3 should be fine:
```diff
@@ -8,6 +8,8 @@
 import sys
 import unittest
 
+import six
+
 from logstash_async.formatter import LogstashFormatter
 
 
@@ -22,9 +24,10 @@
 
 class LogstashFormatterTest(unittest.TestCase):
     def test_format(self):
+        test_message = six.u("тест")
         file_handler = ExceptionCatchingFileHandler(os.devnull)
         file_handler.setFormatter(LogstashFormatter(ensure_ascii=False))
-        file_handler.emit(makeLogRecord({"msg": "тест"}))
+        file_handler.emit(makeLogRecord({"msg": test_message}))
         file_handler.close()
 
         self.assertIsNone(file_handler.exception)
```